hastexo/common.py

```diff
@@ -302,7 +303,13 @@ def read_from_contentstore(course_key, path):
 def ssh_to(user, ip, key):
     ssh = paramiko.SSHClient()
     ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-    pkey = paramiko.RSAKey.from_private_key(StringIO(key))
+
+    pkey = None
+    for key_type in (RSAKey, Ed25519Key):
+        try:
+            pkey = key_type.from_private_key(StringIO(key))
+        except SSHException:
+            pass
```

Comment:
I'm not sure if this silent `pass` is a good idea. Isn't this a "can't happen" error? I mean, shouldn't the `key` argument, if passed in, _always_ be parseable as either an RSA or an ED25519 key? If that is the case, then I'd argue that you shouldn't catch this, and I'd suggest to remove the `try/except` block, and just throw any `SSHException` up the stack.
 
_If_ you want to allow for `key` being passed in as `None` or the empty string, then maybe do this instead?
```suggestion
    if key:
        for key_type in (RSAKey, Ed25519Key):
            pkey = key_type.from_private_key(StringIO(key))
```
Or am I missing something?

Reply:
We don't know which type of key we have here, while we try to connect, so we should either try both and allow the "wrong" type to fail or we need to determine before which key we have and import accordingly.

Reply:
Meh, I'm an idiot. Complete thinko on my part here. 🫤